akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1435,6 +1435,7 @@ private[stream] object Collect {
     private var groupEmitted = true
     private var finished = false
     private var totalWeight = 0L
+    private var elementCount = 0L
```

Comment:
Skip this counting and just check if the `result()` from the `VectorBuilder` is `nonEmpty` in `emitGroup` instead?